### ISSUE TITLE
PyKCS11.PyKCS11Error: Load (No PKCS11 library specified (set PYKCS11LIB env variable))

### ISSUE THREAD

**Issue by alfonsopdiego** [LudovicRousseau/PyKCS11#103]
# Your system information

* Operating system used: Windows 10
* PyKCS11 version: 1.5.12
* Python version: 3.11.4
* PKCS#11 library used: 

# Please describe your issue in as much detail as possible:
first clone the repository and then do this

Install python3 (and add "C:\Python34;C:\Python34\Scripts" to PATH environment variable)
Install swig (and add swig install folder to PATH environment variable)
Install Visual studio 2010 SDK
To install:

Open "Visual Studio command prompt (2010)"

cd to PyKCS11 folder and run::

> nmake -f Makefile.win32 build
> nmake -f Makefile.win32 install

According to the example in the documentation I have followed all the steps and managed to finish the nmake the message final is """"process Finished processing dependencies for PyKCS11==1.5.12 where I find the dll file to load when I do the import in python""""

then test in python and have a error:

import PyKCS11
pkcs11 = PyKCS11.PyKCS11Lib()

pkcs11.load('path_to_dll')

and where is generate the dll?

""""PyKCS11.PyKCS11Error: Load (No PKCS11 library specified (set PYKCS11LIB env variable))""""

In my enviroment i set PYKCS11LIB to the folder when do the nmake build and install

**Comment by LudovicRousseau** [comment]
PyKCS11 is a **WRAPPER** above an existing PKCS#11 library, something like [OpenSC](https://github.com/OpenSC/OpenSC/wiki).
You need to set PYKCS11LIB to the PKCS#11 library you want to use with PyKCS11.

If you do not already have a PKCS#11 library for your card or token then PyKCS11 will not help.

**Comment by alfonsopdiego** [comment]
You are absolutely right, it is true, thanks for the clarification, at first I did not understand how the usb token worked, after you explained that it is a wrapper I understood it and there I understood that the library is to use the generic libraries with python to be able to connect, I have adjusted my code it and it works perfectly, it is evident that for each model of usb key ring there must be a different library, is that right? Because the one I tried, for example, was test from softhsm dll and, it didn't work for me. Then I found the dll library in C:/Windows/System32 folder, digixnip.dll by Digi4id is the manufacturer of the keychain, thank you very much for the prompt response and clarification, it has helped me to solve the problem!
